src/coreclr/jit/gentree.h

```diff
@@ -6687,6 +6691,31 @@ struct GenTreePutArgSplit : public GenTreePutArgStk
 };
 #endif // FEATURE_ARG_SPLIT
 
+#if defined(TARGET_WASM)
+struct GenTreePutArgType : public GenTreeOp
+{
+private:
+    CorInfoType          m_CorInfoType;
+    CORINFO_CLASS_HANDLE m_ClsHnd;
+
+public:
+    GenTreePutArgType(var_types type, GenTree* op, CorInfoType corInfoType, CORINFO_CLASS_HANDLE clsHnd)
+        : GenTreeOp(GT_PUTARG_TYPE, type, op, nullptr), m_CorInfoType(corInfoType), m_ClsHnd(clsHnd)
```

Comment:
```suggestion
        : GenTreeUnOp(GT_PUTARG_TYPE, type, op), m_CorInfoType(corInfoType), m_ClsHnd(clsHnd)
```
(Together with the inheritance change)